include/nbl/asset/utils/ISPIRVOptimizer.h

```diff
@@ -40,8 +41,8 @@ class ISPIRVOptimizer final : public core::IReferenceCounted
 
     ISPIRVOptimizer(std::initializer_list<E_OPTIMIZER_PASS> _passes) : m_passes(std::move(_passes)) {}
 
-    core::smart_refctd_ptr<ICPUBuffer> optimize(const uint32_t* _spirv, uint32_t _dwordCount) const;
-    core::smart_refctd_ptr<ICPUBuffer> optimize(const ICPUBuffer* _spirv) const;
+    core::smart_refctd_ptr<ICPUBuffer> optimize(const uint32_t* _spirv, uint32_t _dwordCount, system::logger_opt_ptr& logger) const;
+    core::smart_refctd_ptr<ICPUBuffer> optimize(const ICPUBuffer* _spirv, system::logger_opt_ptr& logger) const;
```

Comment:
why take a reference?

its just a pseudo pointer, pass by value (anything less or equal to 8 bytes, pass by value)

Reply:
dont pass a reference TODO @sadiuk

Reply:
fixed it on cpp20 branch